homeassistant/components/scene/__init__.py

```diff
@@ -0,0 +1,76 @@
+"""
+homeassistant.components.scene
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Allows users to set and activate scenes.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/scene/
+"""
+import logging
+from collections import namedtuple
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID, SERVICE_TURN_ON, CONF_PLATFORM)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+
+DOMAIN = 'scene'
+DEPENDENCIES = ['group']
+STATE = 'scening'
+
+CONF_ENTITIES = "entities"
+
+SceneConfig = namedtuple('SceneConfig', ['name', 'states'])
+
+
+def activate(hass, entity_id=None):
+    """ Activate a scene. """
+    data = {}
+
+    if entity_id:
+        data[ATTR_ENTITY_ID] = entity_id
+
+    hass.services.call(DOMAIN, SERVICE_TURN_ON, data)
+
+
+def setup(hass, config):
+    """ Sets up scenes. """
+
+    logger = logging.getLogger(__name__)
+
+    for entry in config:
```

Comment:
I see, I guess you're right. We have a helper method that will extract the keys for you: [`extract_domain_configs`](https://github.com/balloob/home-assistant/blob/dev/homeassistant/helpers/__init__.py#L61-L64).

The value of a config can besides being a list also be a dict. If that is the case just wrap the value in a list.

_Disclaimer: I did not test this code_
```python
    # You are not allowed to mutate the original config so make a copy
    config = dict(config)

    for config_key in extract_domain_configs(config, domain):
        platform_config = config[config_key]
        if not isinstance(platform_config, list):
            platform_config = [platform_config]

        if not any(CONF_PLATFORM in entry for entry in platform_config):
            platform_config = [{'platform': 'homeassistant', 'config': entry}
                               for entry in platform_config]

        config[config_key] = platform_config
```